wweval/R/combine_and_summarize_ww_data.R

```diff
@@ -0,0 +1,413 @@
+      latency = as.numeric(lubridate::ymd(
+        !!this_forecast_date
+      ) - lubridate::ymd(max_date))
+    ) |>
+    dplyr::summarize(
+      mean_latency = mean(latency, na.rm = TRUE)
+    ) |>
+    dplyr::pull(mean_latency)
+
+  avg_sampling_freq <- this_ww_data |>
+    dplyr::group_by(lab_wwtp_unique_id) |>
+    dplyr::arrange(date, desc = TRUE) |>
+    # There are some duplicate dates within a site and lab
+    dplyr::distinct(date) |>
+    dplyr::mutate(
+      prev_date = dplyr::lag(date, 1),
+      diff_time = as.numeric(lubridate::days(
+        difftime(date, prev_date)
+      ), "days")
+    ) |>
+    dplyr::ungroup() |>
+    dplyr::summarize(
+      mean_collection_freq = mean(diff_time, na.rm = TRUE)
+    ) |>
+    dplyr::pull(mean_collection_freq)
+
+  n_duplicate_obs <- this_ww_data |>
+    dplyr::group_by(lab_wwtp_unique_id, date) |>
+    dplyr::summarize(n_obs = dplyr::n()) |>
+    dplyr::ungroup() |>
+    dplyr::summarize(
+      n_duplicates = sum(n_obs > 1)
+    ) |>
+    dplyr::pull(n_duplicates)
+
+  this_ww_metadata <- tibble::tibble(
+    forecast_date = this_forecast_date,
+    location = this_location,
+    ww_data_present = 1,
+    n_sites,
+    n_labs,
+    pop_coverage,
+    state_pop,
+    avg_latency,
+    avg_sampling_freq,
+    n_duplicate_obs
+  )
+
+  return(this_ww_metadata)
+}
+
+#' Get additional wastewater metadata
+#'
+#' @description
+#' This function takes the summary statistics from just the input wastewater
+#' data (granular_ww_metadata) and joins the downstream metadata about
+#' convergenece and manual exclusions and combines them all into one table
+#'
+#'
+#' @param granular_ww_metadata a tibble with a row for each forecast date
+#' location and columns that provide summaries of the wastewater data. This
+#' is focused on input data. It is the output of
+#' "combine_and_summarize_ww_data".
+#' @param ww_forecast_date_locs_to_excl table of forecast date-locations
+#' to manually exclude
+#' @param convergence_df table containing convergence flags for every forecast
+#' date location combination
+#' @param table_of_loc_dates_w_ww table containing wastewater metadata
+#' for every location-forecast date with wastewater
+#'
+#' @return a tibble with a number of additional columns indicating whether
+#' or not there were any flags for manual exclusions, convergence issues,
+#' or wastewater quality issues
+#' @export
+get_add_ww_metadata <- function(granular_ww_metadata,
+                                ww_forecast_date_locs_to_excl,
+                                convergence_df,
+                                table_of_loc_dates_w_ww) {
+  granular_ww_metadata_used <- granular_ww_metadata |>
+    dplyr::mutate(forecast_date = lubridate::ymd(forecast_date)) |>
+    dplyr::left_join(
+      ww_forecast_date_locs_to_excl |>
+        mutate(
+          ww_exclude_manual = 1,
+          forecast_date = lubridate::ymd(forecast_date)
+        ),
+      by = c("location", "forecast_date")
+    ) |>
```

Comment:
As written this will have `ww_exclude_manual = NA` whenever data was _not_ manually excluded. I think `ww_exclude_manual = FALSE` makes more sense in that case

```suggestion
    dplyr::left_join(
      ww_forecast_date_locs_to_excl |>
        mutate(
          ww_exclude_manual = TRUE,
          forecast_date = lubridate::ymd(.data$orecast_date)
        ),
      by = c("location", "forecast_date")
    ) |>
    dplyr::mutate(
       ww_exclude_manual = dplyr::replace_na(.data$ww_exclude_manual, FALSE)
    ) |>
```